frontend/src/lib/components/BranchButton.svelte

```diff
@@ -0,0 +1,384 @@
+
+			cache.flush();
+
+			// After updating the tree, check if there's a current file
+			const currentFilePath = get(currentFile);
+			if (currentFilePath) {
+				// Fetch the content of the current file
+				const fileContentResponse = await fetch(
+					`${apiAddress}/api/doc?path=${encodeURIComponent(currentFilePath)}`,
+					{
+						method: 'GET',
+						credentials: 'include'
+					}
+				);
+
+				if (fileContentResponse.ok) {
+					const fileContent = await fileContentResponse.json(); // Get the content of the file
+					editorText.set(fileContent.contents); // Update the editor text
+					cache.set(currentFilePath, fileContent.contents);
+				} else {
+					console.error(
+						'Failed to fetch the file content:',
+						fileContentResponse.status,
+						fileContentResponse.statusText
+					);
+				}
+			}
+		} else {
+			console.error(
+				'Failed to fetch updated document tree:',
+				treeResponse.status,
+				treeResponse.statusText
+			);
+		}
+	}
+
+	function toggleMenu() {
+		showMenu = !showMenu;
+	}
+
+	function closeMenu() {
+		showMenu = false;
+		showInput = false;
+	}
+</script>
+
+<div class="branch-dropdown">
+	<button on:click={toggleMenu} class="branch-button" title="Set Branch Name">
+		{$currentBranch.length > 100 ? `${$currentBranch.slice(0, 100)}...` : $currentBranch}
+	</button>
+
+	{#if showMenu}
+		<div class="branch-menu">
+			<button class="close-button" on:click={closeMenu} aria-label="Close menu">✖</button>
+			<h4>Select Existing Branch</h4>
+			<ul class="branch-list">
+				{#each nonProtectedBranches as branch}
+					<li>
+						<button
+							class="branch-option"
+							on:click={() => setBranchName(branch)}
+							on:keydown={(e) => e.key === 'Enter' && setBranchName(branch)}
+							aria-label={`Select branch ${branch}`}
+						>
+							{branch}
+						</button>
+					</li>
+				{/each}
+				{#if nonProtectedBranches.length === 0}
+					<li>No branches available</li>
+				{/if}
+
+				<!-- "+" button to create a new branch -->
+				<li>
+					{#if !showInput}
+						<button
+							class="branch-option"
+							on:click={() => {
+								showInput = true;
+								newBranchName = '';
+							}}
+							aria-label="Create new branch"
+						>
+							+
+						</button>
+					{:else}
+						<input
+							type="text"
+							bind:value={newBranchName}
+							on:keydown={(e) => {
+								if (e.key === 'Enter') {
+									setBranchName(newBranchName); // Call the function to set the branch name
+									newBranchName = ''; // Reset input field after setting the branch name
+									showInput = false; // Hide input after creating
+								}
+							}}
+							placeholder="Enter new branch name"
+							class="new-branch-input"
+						/>
+					{/if}
+				</li>
+			</ul>
+		</div>
+	{/if}
+</div>
+
+<style>
+	.branch-dropdown {
+		position: relative;
+	}
+
+	.branch-button {
+		position: relative;
+		background-color: var(--button-background);
+		color: var(--button-text);
```

Comment:
I would prefer if these used the foreground or background colorschemes, feel free to add more if needed.

I don't think we should have a hardcoded button color because buttons shouldn't necessarily all the the same color, different colors can be used to convey different levels of importance, or a category of action